### PR TITLE
Changes and improvements for performance and more maintenable code.

### PR DESCRIPTION
**Pull request by meletev**

### CODE REVIEW

**Review by bertrand-lorentz — CHANGES_REQUESTED** [review]
With this change, the unit tests in NoticeViewerTest now all fail on my (Windows) laptop.
I see the unit tests are OK in the GitHub action, and looking at the files for those tests, it seems there is a difference in line endings (CRLF versus LF).

Before this PR, the files were not compared.
I don't think doing an exact comparison is needed or useful: any difference would be mostly caused by a change in the EFX toolkit (so not in the scope of unit tests for this project).
The unit tests are there to check that there is XSL or HTML generated, so some simple sanity check would be enough.

File: src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import net.sf.saxon.trace.TimingTraceListener;
+
+public class HtmlGenerator {
+  private static final Logger logger = LoggerFactory.getLogger(HtmlGenerator.class);
+
+  private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+
+  private final String sdkVersion;
+  private final Path sdkRoot;
+  private Charset charset;
```
Comment (rouschr) [inline src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java:37]: Could be final.

File: src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  private boolean profileXslt;
```
Comment (rouschr) [inline src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java:38]: Could be final.

**Review by rouschr — APPROVED** [review]
Looks OK but see my comments.

File: src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java (hunk 1 above)
Comment (meletev) [reply src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java]: Done

File: src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java (hunk 2 above)
Comment (meletev) [reply src/main/java/eu/europa/ted/eforms/viewer/generator/HtmlGenerator.java]: Done

**Comment by meletev** [comment]
The usage of expected results has been reverted.
As explained by Bertrand, it might be hard to maintain and it also has newline issues on Windows, which can be re-introduced even if fixed now.

**Review by rouschr — APPROVED** [review]

**Review by rousso — DISMISSED** [review]

**Review by rousso — APPROVED** [review]

**Pull request merged**